dev/bots/analyze.dart

```diff
@@ -161,11 +159,23 @@ Future<void> run(List<String> arguments) async {
 
   // Analyze all the Dart code in the repo.
   printProgress('Dart analysis...');
-  await _runFlutterAnalyze(flutterRoot, options: <String>[
+  final CommandResult analyzeResults = await _runFlutterAnalyze(flutterRoot, options: <String>[
     '--flutter-repo',
     ...arguments,
   ]);
 
+  if (analyzeResults.exitCode == 0) {
+    // Only run the private lints when the code is free of type errors. The
+    // lints are easier to write when they can assume, for example, there is no
+    // inheritance cycles.
+    final List<ResolvedUnitVerifier> verifiers = <ResolvedUnitVerifier>[verifyNoDoubleClamp, verifyDebugAssertAccess];
+    final String rules = verifiers.map((ResolvedUnitVerifier verifier) => '\n * $verifier').join();
+    printProgress('Analyzing code in the framework with the following rules:$rules');
+    await runVerifiersInResolvedDirectory(flutterRoot, verifiers);
+  } else {
+    printProgress('Skipped analyzing code in the framework because dart analysis finished with a non-zero exit code.');
```

Comment:
This reads a little awkward: Skipped analyzing because analyzing failed? So, how does it know that analysis failed if it skipped it? 😆 

Maybe be a little more explicit about WHAT analysis was skipped?

Reply:
"Skipped performing further analysis in the framework because "flutter analyze" finished with a non-zero exit code."?

Reply:
Sure, don't you have the names of the verifiers that were supposed to run? Optionally you could say that here to tell people exactly what wasn't checked.